src/gui/include/gui/gui.h

```diff
@@ -269,7 +269,9 @@ class Descriptor
   {
     return getTypeName();
   }
-  virtual bool getBBox(std::any object, odb::Rect& bbox) const = 0;
+  virtual bool getBBox(std::any object,
+                       std::any additional_data,
+                       odb::Rect& bbox) const = 0;
```

Comment:
What is the purpose of additional data here?

Reply:
The additional_data for the dbRow is storing the bbox of the site or row, it's needed there so it can be used by the descriptors. I thought you wanted to be able to select rows and sites, so I made it such that you can select a site in a row when zoomed far enough in and just the row when zoomed less in. If what you were going for was just the row, then I can revert the saving of the Rect's for dbRow.

Reply:
I see the site needs to have this as it doesn't have a bbox of its own.  That makes sense.